### DISCUSSION TITLE
Using `@component layer` with `postcss-import` and `create-react-app`.

### DISCUSSION THREAD

**Discussion by rizkimuhammada** [tailwindlabs/tailwindcss#8259] [Help]
I just created a fresh react app with `create-react-app`, following the steps in the documentation.
https://tailwindcss.com/docs/guides/create-react-app

Then try using `postcss-import` and `@layer components` directives. But I keep getting errors.

> `@layer components` is used but no matching `@tailwind components` directive is present.

Here is the CSS code.

**App.css**
```
@tailwind base;
@tailwind components;
@tailwind utilities;

@import "Example.css";
```

**Example.css**
```
// Error
@layer components {
  header {
    @apply bg-red-500;
  }
}

// This works (without @layer components)
header {
  @apply bg-red-500;
}
```

**postcss.config.js**
```
module.exports = {
  plugins: {
    "postcss-import": {},
    tailwindcss: {},
    autoprefixer: {},
  },
};
```

I tried using `vite` and didn't find this problem, is it a known issue?

**Comment by andrewbiddinger** [comment]
I'm getting the same issue with the exact setup as well.

**Comment by andrewbiddinger** [comment]
So, I tried adding `@import "tailwindcss/components";` above `@layer components {` which was recommended on this issue: https://github.com/tailwindlabs/tailwindcss/discussions/5934 and now I can build the app. But, I'm still investigating why this is needed.

**Reply by andrewbiddinger** [reply]
I think this problem is more related to Tailwind not playing very well with Create React app/postcss-import. 

> Create React App does not support custom PostCSS configurations and is incompatible with many important tools in the PostCSS ecosystem, like `postcss-import`.
> 
> We highly recommend using [Vite](https://tailwindcss.com/docs/guides/vite), [Parcel](https://tailwindcss.com/docs/guides/parcel), [Next.js](https://tailwindcss.com/docs/guides/nextjs), or [Remix](https://tailwindcss.com/docs/guides/remix) instead of Create React App. They provide an equivalent or better developer experience but with more flexibility, giving you more control over how Tailwind and PostCSS are configured.
